ProductDock/rbc-library-ui
# Enable MSW (mock service worker)

### What did you implement?

Mocking of API calls by intercepting them on the network level. Seamlessly reuse the same mock definition for testing, development, and debugging.

- [x] Integrate `MSW` with browser
- [x] Integrate `MSW` with tests 
- [x] Centralize mocking of authentication context
- [x] Run all tests on git push

### Overview

The `msw` folder is placed under the `src` folders, and contains the following 3 files:

- `handlers.ts` - contains request handlers (mocked APIs) that can be shared between browser and server/node environment.
- `browser.ts` - used only if you want to mock requests during development in the browser.
- `server.ts` - used for tests only, contains shared handlers and server setup.

### Steps to test your changes
- `npm install`
- `.env` -> `REACT_APP_MSW_MOCKING_ENABLED=true`
- `npm start`
- Check if the books are rendered on the page.
